lib/shopify-cli/commands.rb

```diff
@@ -7,23 +7,50 @@ module Commands
       contextual_resolver: nil,
     )
 
-    def self.register(const, cmd, path = nil)
-      autoload(const, path) if path
-      Registry.add(->() { const_get(const) }, cmd)
+    class << self
+      def register(const, cmd, path = nil)
+        autoload(const, path) if path
+        Registry.add(->() { const_get(const) }, cmd)
+      end
+
+      def load_commands_for_all_contexts
```

Comment:
So in order to take advantage of context-specific commands you need to load them in one of these named methods _and_ use the `available_in_contexts` classmethod in your command class. Isn't defining contextuality in two different places very confusing? In #439 this is all handled by the `available_in` method and you don't need to conditionally autoload anything.